.github/workflows/publish.yml

```
@@ -0,0 +1,44 @@
+name: Publish
+on:
+  push:
+    tags:
+      - '*'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.11
+
+    - name: Install dependencies and build
+      run: |
+        python3 -m pip install --upgrade pip
+        pip install poetry
+        poetry build
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        path: ./dist
+
+  pypi-publish:
+    needs: ['build']
+    environment: 'PypiProd'
+
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+      - uses: actions/download-artifact@v3
```

```suggestion
      - name: Download artifacts
        uses: actions/download-artifact@v3
```
I like to name all my steps.  It just makes the output logs a bit easier to read.